File: gateway/src/main/java/com/oligei/gateway/service/AuctionService.java
package com.oligei.gateway.service;

import com.oligei.gateway.dto.AuctionListItem;
import com.oligei.gateway.util.msgutils.Msg;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@FeignClient(value = "auction-feign")
public interface AuctionService {

    @RequestMapping(value = "/Auction/add",method = RequestMethod.POST)
    Msg<Boolean> addAuction(@RequestParam("actitemid")Integer actitemid,
                                   @RequestParam("ddl")String ddl,
                                   @RequestParam("showtime")String showtime,
                                   @RequestParam("initprice")Integer initprice,
                                   @RequestParam("orderprice")Integer orderprice,
                                   @RequestParam("amount")Integer amount);


    @RequestMapping(value = "/Auction/get",method = RequestMethod.GET)
    Msg<Map<Integer,AuctionListItem>> getAuctions();

    @RequestMapping(value = "/Auction/join",method = RequestMethod.POST)
    Msg<Integer> joinAuction(@RequestParam("auctionid") java.lang.Integer auctionid,
                                          @RequestParam("userid") java.lang.Integer userid,
                                          @RequestParam("price") java.lang.Integer price);

    @RequestMapping("/Auction/canEnter")
    Msg<Boolean> canEnter(@RequestParam("userid") Integer userid, @RequestParam("auctionid") Integer auctionid);

    @RequestMapping("/Auction/deposit")
    Msg<Boolean> deposit(@RequestParam("userid") Integer userid, @RequestParam("auctionid") Integer auctionid);

    @RequestMapping("/Auction/getPrice")
    Msg<Integer> getPrice(@RequestParam("auctionid") Integer auctionid);
}

File: frontend/src/service/AuctionService.js
import {authRequest, postRequest} from "../utils/ajax";

export const getAuctions = async (token, callback) => {
    const data = {searching: null};
    const url = `/auction/get`;
    await authRequest(url, data, token, callback);
};

export const addAuction = async (actitemid, ddl, showtime, initprice, orderprice, amount, token, callback) => {
    const data = {
        actitemid: actitemid,
        ddl: ddl,
        showtime: showtime,
        initprice: initprice,
        orderprice: orderprice,
        amount: amount
    };
    const url = `/auction/add`;
    console.log(data);
    await authRequest(url, data, token, callback);
};

export const joinAuctions = async (auctionid, userid, price, token, callback) => {
    const data = {auctionid: auctionid, userid: userid, price: price};
    const url = `/auction/join`;
    await authRequest(url, data, token, callback);
};

export const enterAuction= async (userId, auctionId, token, callback) => {
    const data = {userid: userId, auctionid: auctionId};
    const url = `/auction/canEnter`;
    await authRequest(url, data, token, callback);
}

export const deposit= async (userId, auctionId, token, callback) => {
    const data = {userid: userId, auctionid: auctionId};
    const url = `/auction/deposit`;
    await authRequest(url, data, token, callback);
}

export const getPriceById= async (auctionId, token, callback) => {
    const data = {auctionid: auctionId};
    const url = `/auction/getPrice`;
    await authRequest(url, data, token, callback);
}
